### ISSUE TITLE
[Bug]: Button on table does not behave similar to button widget on canvas

### ISSUE THREAD

**Issue by shwetha-ramesh** [appsmithorg/appsmith#10278]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When button label is long there is no ```ellipsis...``` seen in table widget. Where as a button widget on canvas, form or a container when the label is long an ellipsis is added next to the text. 
Also with a long button label when a new column is added next to it, the button gets hidden partially.

![image](https://user-images.githubusercontent.com/92293815/148754925-c486438f-f40f-443d-b80e-c27a9cb94e5e.png)

https://loom.com/share/19d7274a1c164535ace3ce21e7998cbf

Expected behavior: Consistency to be maintained across all

### Steps To Reproduce

1. Drag and drop a table widget
2. Click on setting icons next to _action_ column
3. Change the Button label to "Naming this button for  a long name for testing" and observe

Follow the above steps for Button widget when placed on canvas and observe. 

### Environment

Production

### Version

Cloud

**Comment by somangshu** [comment]
@sbalaji1192 do you think we can resize the button based on the column size? What other bottelnecks you can think of in this implementation

**Comment by sbalaji1192** [comment]
@somangshu Yeah. we have the column size details in the component. so we can set the width based on the column size.
We have plans to use widget components for buttons, icon buttons, etc in the table as well. I guess this will get resolved as part of those changes.

**Comment by dilippitchika** [comment]
Needs more discussion around this

Suggested next steps:
1. Start with using the same button widget in the table 
2. Observe the discrepancies and then resolve
Needs a small discussion on this before we pick this up due to a bigger impact due to the changes. Needs to be tested well enough in the next version of table.

## Stats
| Stat | Values |
| ------------- | ------------- |
| Reach     | 1000 |
| Effort (months)  | 1.25 |

**Comment by Aishwarya-U-R** [comment]
Users also need ability to set up 'Tooltip' to Table widget Button type columns.

**Comment by somangshu** [comment]
@Aishwarya-U-R thanks for the point here. Since we are trying to make it consistent all properties of the button should be visible here.

**Comment by dvj1988** [comment]
@rajatagrawal I am picking this up as part of DMI this sprint.

**Comment by carinanfonseca** [comment]
Similar to this one: https://github.com/appsmithorg/appsmith/issues/17940

**Comment by skjameela** [comment]
@carinanfonseca i would like to work on this

**Comment by skjameela** [comment]
@somangshu for this do i need to add tooltip also or elipsis is enough?

**Comment by skjameela** [comment]
@rahulbarwal  do i need to add tooltip for the button widget component ?

**Comment by rahulbarwal** [comment]
No, normal button widget component already has a tooltip. We need to have similar functionality in button inside table column.
![Screenshot 2024-10-07 at 5 25 49 PM](https://github.com/user-attachments/assets/337d6490-8b38-4870-be30-485992a64397)

**Comment by skjameela** [comment]
@rahulbarwal ok got it .  in table coulmn for the overflow content and url tooltip already implemented like that i have been implemented for button content as well is that okay ?

**Comment by rahulbarwal** [comment]
@skjameela Yes, correct. To clarify and conclude.
There are 2 requirements here, both are for buttons inside table column. The normal button widget is fine(and already has both features)
1. ellipsis for overflowing content.
2. tooltip functionality.
